test/integration/executor/add_asset_qty_test.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "integration/executor/executor_fixture.hpp"
+
+#include <gtest/gtest.h>
+#include "cryptography/crypto_provider/crypto_defaults.hpp"
+#include "framework/common_constants.hpp"
+#include "framework/result_gtest_checkers.hpp"
+#include "integration/executor/executor_fixture_param_provider.hpp"
+#include "interfaces/common_objects/amount.hpp"
+#include "module/shared_model/mock_objects_factories/mock_command_factory.hpp"
+#include "module/shared_model/mock_objects_factories/mock_query_factory.hpp"
+
+using namespace common_constants;
+using namespace executor_testing;
+using namespace framework::expected;
+
+using shared_model::interface::permissions::Grantable;
+using shared_model::interface::permissions::Role;
+
+using shared_model::interface::Amount;
+
+static const Amount kAmount{std::string{"12.3"}};
+
+using AddAssetQuantityBasicTest =
+    executor_testing::BasicExecutorTest<executor_testing::ExecutorTestBase>;
+
+/**
+ * @given an asset in another domain and a user with kAddAssetQty permission
+ * @when execute AddAssetQuantity command from that user for that asset
+ * @then the command succeeds
+ * @and the asset quantity gets increased
+ */
+TEST_P(AddAssetQuantityBasicTest, Basic) {
+  getItf().createDomain(kSecondDomain);
+  createAsset(kAssetName, kSecondDomain, 1);
+  assertResultValue(getItf().createUserWithPerms(
+      kUser, kDomain, kUserKeypair.publicKey(), {Role::kAddAssetQty}));
+
+  expectResultValue(getItf().executeCommandAsAccount(
+      *getItf().getMockCommandFactory()->constructAddAssetQuantity(
+          kAnotherDomainAssetId, kAmount),
+      kUserId,
+      true));
```

Comment:
`true` looks confusing to me, and I am not sure whether we need to execute any command without permission checks. Maybe add a TODO to make a more clear interface?

Reply:
I think we should only validate the commands that are part of test fulfilment criteria. The rest can go without validation if it makes them simpler.